app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -112,21 +122,26 @@ class SmartTransactionHook {
       isSmartTransaction,
       controllerMessenger,
       featureFlags,
+      transactions,
+      chainId,
     } = request;
     this.#approvalFlowId = '';
     this.#approvalFlowEnded = false;
-    this.#transactionMeta = transactionMeta;
+    this.#transactionMeta = transactionMeta as TransactionMeta;
     this.#signedTransactionInHex = signedTransactionInHex;
     this.#smartTransactionsController = smartTransactionsController;
     this.#transactionController = transactionController;
     this.#isSmartTransaction = isSmartTransaction;
     this.#controllerMessenger = controllerMessenger;
     this.#featureFlags = featureFlags;
-    this.#isDapp = transactionMeta.origin !== ORIGIN_METAMASK;
-    this.#chainId = transactionMeta.chainId;
-    this.#txParams = transactionMeta.txParams;
-    this.#shouldShowStatusPage =
-      transactionMeta.type !== TransactionType.bridge;
+    this.#isDapp = transactionMeta?.origin !== ORIGIN_METAMASK;
```

Comment:
TODO: Since we don't pass `transactionMeta` in the submitBatch hook, we need to update this. We can perhaps pass the `origin` or `isDapp` in the constructor.

Reply:
Resolved by always passing `transactionMeta` into the hook.